### PR TITLE
[#2980] test(core):  add  test for event listener

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add test  for event 

### Why are the changes needed?

Fix: #2980 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
test patch

### CODE REVIEW

**Comment by jerryshao** [comment]
I think you don't have to split into multiple small subtasks, you can add all in one PR.

**Comment by FANNG1** [comment]
> I think you don't have to split into multiple small subtasks, you can add all in one PR.

ok

File: core/src/test/java/com/datastrato/gravitino/listener/DummyEventListener.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public List<Event> tryGetEvents() {
+      Instant waitTime = Instant.now().plusSeconds(20);
+      while (getEvents().size() == 0 && Instant.now().isBefore(waitTime)) {
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
+      return getEvents();
```
Comment (jerryshao) [inline core/src/test/java/com/datastrato/gravitino/listener/DummyEventListener.java:57]: You can use awaitability to improve your code here, no need to write it manually.

File: core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    this.dummyEventListener = new DummyEventListener();
+    EventBus eventBus = new EventBus(Arrays.asList(dummyEventListener));
+    CatalogDispatcher catalogDispatcher = mockCatalogDispatcher();
+    this.dispatcher = new CatalogEventDispatcher(eventBus, catalogDispatcher);
+    CatalogDispatcher catalogExceptionDispatcher = mockExceptionCatalogDispatcher();
+    this.failureDispatcher = new CatalogEventDispatcher(eventBus, catalogExceptionDispatcher);
+  }
+
+  @Test
+  void testCreateCatalog() {
```
Comment (jerryshao) [inline core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java:50]: I think you'd better rename the method to `testCreateCatalogEvent` here and below? Actually, you're testing the event, not the create operation, right?

Comment (FANNG1) [reply core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java:50]: yes

File: core/src/test/java/com/datastrato/gravitino/listener/DummyEventListener.java (hunk 1 above)
Comment (FANNG1) [reply core/src/test/java/com/datastrato/gravitino/listener/DummyEventListener.java:57]: ok

Comment (FANNG1) [reply core/src/test/java/com/datastrato/gravitino/listener/DummyEventListener.java]: done

File: core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java (hunk 2 above)
Comment (FANNG1) [reply core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java]: done

File: core/build.gradle.kts [hunk 3]
```diff
@@ -37,6 +37,7 @@ dependencies {
   testImplementation(libs.junit.jupiter.api)
   testImplementation(libs.junit.jupiter.params)
   testImplementation(libs.mockito.core)
+  testImplementation(libs.testcontainers)
```
Comment (jerryshao) [inline core/build.gradle.kts:40]: Why do we need to introduce this?

Comment (FANNG1) [reply core/build.gradle.kts:40]: I used the awaitability from `testcontainers`,  will use from `awaitility` instread

**Review by jerryshao — APPROVED** [review]

**Pull request merged**